### PR TITLE
__mul__ method for CSR

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented __mul__ operator for CSR matrix.
merge #238 first
related to #36

### CODE REVIEW

**Comment by unnonouno** [comment]
I added tests for shapes and fixed docstring.

File: cupy/cusparse.py [hunk 1]
```diff
@@ -45,6 +52,158 @@ def _call_cusparse(name, dtype, *args):
...
+    .. math::
+
+       C = \\alpha o_a(A) B + \\beta C,
+
+    where :math:`o_a` is a transpose function when ``transa`` is ``True`` and
+    is an identity function otherwise.
+
+    Args:
+        a (cupy.sparse.csr): Sparse matrix A.
+        b (cupy.ndarray): Dense matrix B. It must be F-contiguous.
```
Comment (takagi) [inline cupy/cusparse.py]: Requiring F-contiguousness as the function's spec? `cupy.asfortranarray()` is called inside.

File: cupy/cusparse.py [hunk 2]
```diff
@@ -45,6 +52,158 @@ def _call_cusparse(name, dtype, *args):
...
+        c (cupy.ndarray or None): Dense matrix C. It must be F-contiguous.
```
Comment (takagi) [inline cupy/cusparse.py]: ditto

File: cupy/cusparse.py [hunk 3]
```diff
@@ -45,6 +52,158 @@ def _call_cusparse(name, dtype, *args):
...
+
+       C = \\alpha o_a(A) o_b(B) + \\beta C,
+
+    where :math:`o_a` and :math:`o_b` are transpose functions when ``transa``
+    and ``tranb`` are ``True`` respectively. And they are identity functions
+    otherwise.
+
+    Args:
+        a (cupy.sparse.csr): Sparse matrix A.
+        b (cupy.ndarray): Dense matrix B. It must be F-contiguous.
```
Comment (takagi) [inline cupy/cusparse.py]: ditto

File: cupy/cusparse.py [hunk 4]
```diff
@@ -45,6 +52,158 @@ def _call_cusparse(name, dtype, *args):
...
+        c (cupy.ndarray or None): Dense matrix C. It must be F-contiguous.
```
Comment (takagi) [inline cupy/cusparse.py]: ditto

File: cupy/cusparse.py [hunk 5]
```diff
@@ -45,6 +52,160 @@ def _call_cusparse(name, dtype, *args):
...
+
+       y = \\alpha * o_a(A) x + \\beta y,
+
+    where :math:`o_a` is a transpose function when ``transa`` is ``True`` and
+    is an identity function otherwise.
+
+    Args:
+        a (cupy.cusparse.csr_matrix): Matrix A.
+        x (cupy.ndarray): Vector x.
+        y (cupy.ndarray or None): Vector y. It must be contiguous.
```
Comment (takagi) [inline cupy/cusparse.py]: It would be better to note F-contiguous.

**Comment by takagi** [comment]
Would you check the CI result?

**Comment by unnonouno** [comment]
fixed

**Pull request merged**